Backend/ssointegration /src/main/java/com/ssointegration/controller/UserController.java

```diff
@@ -0,0 +1,85 @@
+package com.ssointegration.controller;
+
+import com.ssointegration.dto.Auth;
+import com.ssointegration.dto.UserDto;
+import com.ssointegration.service.JwtService;
+import com.ssointegration.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Collections;
+import java.util.List;
+
+
+@RestController
+@RequestMapping("/users")
+public class UserController {
+    @Autowired
+    private UserService userService;
+
+    @Autowired
+    private JwtService jwtService;
+
+    @GetMapping
+    public ResponseEntity<List<UserDto>> findAllUsers(@RequestHeader("Authorization") String authorizationHeader) {
+        try {
+            String token = extractTokenFromAuthorizationHeader(authorizationHeader);
+
+            jwtService.validateToken(token);
+
+            List<UserDto> users = userService.getAllUsers();
+            return new ResponseEntity<>(users, HttpStatus.OK);
+        } catch (Exception e) {
+            return ResponseEntity.status(HttpStatus.UNAUTHORIZED).body(Collections.emptyList());
+        }
+    }
+
+    @PostMapping
+    public ResponseEntity<UserDto> saveUser(@RequestBody UserDto userDto, @RequestHeader("Authorization") String authorizationHeader) {
```

Comment:
don't fetch the token like this. you have to fetch it in the JwtService class that you created or do it in the filter